changing Debian pidfile

colinhoglund opened the pull request:
The packaged version of nginx.conf sets the pid file to /run/nginx.pid. When the pid value is changed and nginx receives the reload signal, the pid file gets removed. This causes 'service: name=nginx state=started enabled=yes' to think nginx is down on subsequent ansible runs.
See https://trac.nginx.org/nginx/ticket/796 for more info.

This PR creates a 'nginx_pidfile' role var in order to change the nginx.conf.j2 pid for Debian hosts.

geerlingguy commented:
@colinhoglund - The `/run` directory is nonstandard (see: https://www.debian.org/releases/squeeze/i386/apcs02.html.en)... and I don't want to break the default filesystem conventions with the role defaults... can we leave the default as `/var/run/nginx.pid` and then playbooks can override if needed?

colinhoglund commented:
Thanks for the reply, that is a good point. I have verified that the /run directory exists on Ubuntu 12.04/14.04 and Debian 7/8. Also, after reading your comment I did some research and stumbled across an interesting email that explains the standardization of the /run directory. http://thread.gmane.org/gmane.linux.redhat.fedora.devel/146976

As a user, I would not expect I need to override the default pid file for a service. Using the role in its current state can result in the removal of the nginx pid file, breaking the associated init script, and in some cases causing later ansible runs to spin up multiple nginx master processes (see following example). My recommendation would be to use the packaged default of /run/nginx.pid in order to avoid confusion.

test.yml
```
---

- hosts: localhost
  connection: local
  roles:
    - ansible-role-nginx
```
3 consecutive runs of test.yml on a fresh ubuntu 14.04 install
```
root@vagrant-ubuntu-trusty-64:/tmp# ansible-playbook test.yml
 [WARNING]: provided hosts list is empty, only localhost is available


PLAY ***************************************************************************

TASK [setup] *******************************************************************
ok: [localhost]

TASK [ansible-role-nginx : Include OS-specific variables.] *********************
ok: [localhost]

TASK [ansible-role-nginx : Define nginx_user.] *********************************
ok: [localhost]

TASK [ansible-role-nginx : include] ********************************************
skipping: [localhost]

TASK [ansible-role-nginx : include] ********************************************
included: /tmp/roles/ansible-role-nginx/tasks/setup-Debian.yml for localhost

TASK [ansible-role-nginx : Update apt cache.] **********************************
ok: [localhost]

TASK [ansible-role-nginx : Ensure nginx is installed.] *************************
changed: [localhost]

TASK [ansible-role-nginx : include] ********************************************
included: /tmp/roles/ansible-role-nginx/tasks/vhosts.yml for localhost

TASK [ansible-role-nginx : Remove default nginx vhost config file (if configured).] ***
skipping: [localhost]

TASK [ansible-role-nginx : Add managed vhost config file (if any vhosts are configured).] ***
skipping: [localhost]

TASK [ansible-role-nginx : Remove managed vhost config file (if no vhosts are configured).] ***
ok: [localhost]

TASK [ansible-role-nginx : Copy nginx configuration in place.] *****************
changed: [localhost]

TASK [ansible-role-nginx : Ensure nginx is started and enabled to start at boot.] ***
ok: [localhost]

RUNNING HANDLER [ansible-role-nginx : reload nginx] ****************************
changed: [localhost]

PLAY RECAP *********************************************************************
localhost                  : ok=11   changed=3    unreachable=0    failed=0

root@vagrant-ubuntu-trusty-64:/tmp# service nginx status
 * nginx is not running

root@vagrant-ubuntu-trusty-64:/tmp# ps -ef | grep nginx
root      7458     1  0 00:54 ?        00:00:00 nginx: master process /usr/sbin/nginx
www-data  7573  7458  0 00:54 ?        00:00:00 nginx: worker process
root      7598  1639  0 00:55 pts/0    00:00:00 grep --color=auto nginx

root@vagrant-ubuntu-trusty-64:/tmp# ansible-playbook test.yml
 [WARNING]: provided hosts list is empty, only localhost is available


PLAY ***************************************************************************

TASK [setup] *******************************************************************
ok: [localhost]

TASK [ansible-role-nginx : Include OS-specific variables.] *********************
ok: [localhost]

TASK [ansible-role-nginx : Define nginx_user.] *********************************
ok: [localhost]

TASK [ansible-role-nginx : include] ********************************************
skipping: [localhost]

TASK [ansible-role-nginx : include] ********************************************
included: /tmp/roles/ansible-role-nginx/tasks/setup-Debian.yml for localhost

TASK [ansible-role-nginx : Update apt cache.] **********************************
ok: [localhost]

TASK [ansible-role-nginx : Ensure nginx is installed.] *************************
ok: [localhost]

TASK [ansible-role-nginx : include] ********************************************
included: /tmp/roles/ansible-role-nginx/tasks/vhosts.yml for localhost

TASK [ansible-role-nginx : Remove default nginx vhost config file (if configured).] ***
skipping: [localhost]

TASK [ansible-role-nginx : Add managed vhost config file (if any vhosts are configured).] ***
skipping: [localhost]

TASK [ansible-role-nginx : Remove managed vhost config file (if no vhosts are configured).] ***
ok: [localhost]

TASK [ansible-role-nginx : Copy nginx configuration in place.] *****************
ok: [localhost]

TASK [ansible-role-nginx : Ensure nginx is started and enabled to start at boot.] ***
changed: [localhost]

PLAY RECAP *********************************************************************
localhost                  : ok=10   changed=1    unreachable=0    failed=0

root@vagrant-ubuntu-trusty-64:/tmp# service nginx status
 * nginx is running

root@vagrant-ubuntu-trusty-64:/tmp# ps -ef | grep nginx
root      7458     1  0 00:54 ?        00:00:00 nginx: master process /usr/sbin/nginx
www-data  7573  7458  0 00:54 ?        00:00:00 nginx: worker process
root      8054     1  0 00:55 ?        00:00:00 nginx: master process /usr/sbin/nginx
www-data  8055  8054  0 00:55 ?        00:00:00 nginx: worker process
root      8070  1639  0 00:55 pts/0    00:00:00 grep --color=auto nginx
```
